src/docker-in-docker/install.sh

```diff
@@ -333,65 +332,38 @@ fi
 
 usermod -aG docker ${USERNAME}
 
-# Function to fetch the latest version of the plugin
-get_latest_version() {
-    repo_url=$1
-    curl -s "$repo_url/latest" | jq -r '.tag_name'
-}
-
 # Function to fetch the version released prior to the latest version
 get_previous_version() {
-    latest_version="$1"
-    repo_url=$2
-    # curl -s "$repo_url" | jq -r --arg latest "$latest_version" '.[].tag_name | select(. != $latest)' | sort -rV | head -n 1   # this helps filter tag_names based on semver specifications, this one was incorrectly installing 0.12.0-rc2 which is the third in the list i.e seeing from top to bottom
+    repo_url=$1
     curl -s "$repo_url" | jq -r 'del(.[].assets) | .[1].tag_name' # this would del the assets key and then get the second encountered tag_name's value from the filtered array of objects
 }
 
-# Function to change the patch number in a semver version
-change_patch_number() {
-    local version="$1"  # Input version
-    local new_patch="$2"  # New patch number
-    # Extract major, minor, and current patch numbers
-    local major=$(echo "$version" | cut -d. -f1)
-    local minor=$(echo "$version" | cut -d. -f2)
-    local current_patch=$(echo "$version" | cut -d. -f3)
-    # Construct the new version with the updated patch number
-    local new_version="$major.$minor.$new_patch"
-    echo "$new_version"
+fetch_previous_version() {
+    wget_exit_code=$?
+    if [ $wget_exit_code -ne 0 ]; then # means wget command to fetch latest version failed
+        if [ $wget_exit_code -eq 8 ]; then  # failure due to 404: Not Found.
+            echo -e "\n(!) Failed to fetch the latest artifacts for docker buildx v${buildx_version}..."
```

Comment:
Question: Will `buildx_version` be available to this function or it needs to be passed in?